docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -1078,6 +1079,122 @@ public @interface CanWrite {
+To avoid ambiguous resolution, automatic detection only works for the `beanParam` fields.
+For that reason, we had to specify path to the user principal explicitly.
+
+Where the `SimpleBeanParam` class is declared like in the example below:
+
+[source,java]
+----
+package org.acme.security.rest.dto;
+
+import java.util.List;
+
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.SecurityContext;
+import jakarta.ws.rs.core.UriInfo;
+
+public class SimpleBeanParam {
+
+    @HeaderParam("CustomAuthorization")
+    private String customAuthorizationHeader;
+
+    @Context
+    SecurityContext securityContext;
+
+    @Context
+    public UriInfo uriInfo;
+
+    @QueryParam("query")
+    public String query;    <1>
+
+    public SecurityContext getSecurityContext() { <2>
+        return securityContext;
+    }
+
+    public String customAuthorizationHeader() { <3>
+        return customAuthorizationHeader;
+    }
+}
+----
+<1> Quarkus Security can only pass public fields to a custom permission constructor.
+<2> Quarkus Security automatically uses public getter methods if they are available.
+<3> The `customAuthorizationHeader` field is not public, therefore Quarkus access this field with the `customAuthorizationHeader` accessor.
+That is particularly useful with Java records, where generated accessors are not prefixed with `get`.
+
+Here is an example of the `BeanParamPermission` permission that checks user principal, custom header and query parameter:
+
+[source,java]
+----
+package org.acme.security.permission;
+
+import java.security.Permission;
+
+public class BeanParamPermission extends Permission {
+
+    private final String permissionName;
+    private final String customAuthorization;
+    private final String userName;
+    private final String queryParam;
+
+    public BeanParamPermission(String permissionName, String customAuthorizationHeader, String name, String query) {
+        super(permissionName);
+        this.permissionName = permissionName;
+        this.customAuthorization = customAuthorizationHeader;
+        this.userName = name;
+        this.queryParam = query;
+    }
+
+    @Override
+    public boolean implies(Permission permission) {
+        if (permission instanceof BeanParamPermission that) {
+            boolean permissionNameMatches = permissionName.equals(that.permissionName);
+            boolean queryParamAllowedForPermissionName = checkQueryParams(that.queryParam);
+            boolean usernameWhitelisted = isUserNameWhitelisted(that.userName);
+            boolean customAuthorizationMatches = checkCustomAuthorization(that.customAuthorization);
+            return permissionNameMatches && queryParamAllowedForPermissionName
+                    && usernameWhitelisted && customAuthorizationMatches;
+        }
+        return false;
+    }
+
+    ...
+}
+----
+
```

Comment:
Probably here, I believe there should be a note, even if it may be obvious, that BeanParam beans can be passed directly to the permission constructors, like any other JAX-RS method parameter

Reply:
done, let's see